src/bam/mod.rs

```diff
@@ -155,6 +155,15 @@ impl Reader {
     /// * `path` - the path to open. Use "-" for stdin.
     fn new(path: &[u8]) -> Result<Self, BGZFError> {
         let htsfile = try!(hts_open(&ffi::CString::new(path).unwrap(), b"r"));
+        unsafe {
+            if (*htsfile).format.format != htslib::htsExactFormat_sam
```

Comment:
From a quick glance at [hts_detect_format()](https://github.com/samtools/htslib/blob/5c7ff9c6812f1e6ef1fd83ba73b8e42cfbabba93/hts.c#L199), it looks to me like you could substitute the three checks for `format.format` here with one check for `format.category`, i.e.:
```
if (*htsfile).format.category != htslib::htsFormatCategory_sequence_data
```
This should be more general and hopefully more future-proof, should htslib e.g. invent a newer sequence_data format...

Or am I overlooking something, here?

Reply:
No you're right.